restli-tools/src/main/resources/apiVmTemplates/collection.vm

```diff
@@ -1,4 +1,138 @@
+    import com.linkedin.restli.common.CollectionRequest;
+    #if($method.returnsEntity())
+      import com.linkedin.restli.client.BatchCreateIdEntityRequest;
+      import com.linkedin.restli.common.CreateIdEntityStatus;
+      import com.linkedin.restli.common.BatchCreateIdEntityResponse;
+      import com.linkedin.restli.internal.client.BatchCreateIdEntityDecoder;
+    #else
+     import com.linkedin.restli.client.BatchCreateIdRequest;
+     import com.linkedin.restli.common.CreateIdStatus;
+     import com.linkedin.restli.common.BatchCreateIdResponse;
+     import com.linkedin.restli.internal.client.BatchCreateIdDecoder;
+    #end
+  #end
+#end
+@Generated("Generated from $spec.sourceIdlName")
+public class $spec.className {
+  private final static String ORIGINAL_RESOURCE_PATH = "$util.getResourcePath($spec.resource.path)";
+  private static ResourceSpec _resourceSpec;
+
+  static {
+    HashMap<String, DynamicRecordMetadata> requestMetadataMap = new HashMap<String, DynamicRecordMetadata>();
+    #foreach($action in $spec.actions)
+      ArrayList<FieldDef<?>> ${action.name}Params = new ArrayList<FieldDef<?>>();
+      #foreach($actionParam in $action.parameters)
+        ${action.name}Params.add(new FieldDef<${actionParam.paramClassName}>("${actionParam.paramName}", ${actionParam.paramClassName}.class, DataTemplateUtil.getSchema(${actionParam.paramClassName}.class)));
+      #end
+      requestMetadataMap.put("${action.name}", new DynamicRecordMetadata("${action.name}", ${action.name}Params));
+    #end
+
+    HashMap<String, DynamicRecordMetadata> responseMetadataMap=new HashMap<String, DynamicRecordMetadata>();
+    #foreach($action in $spec.actions)
+      #if($action.schema.hasReturns())
+        responseMetadataMap.put("${action.name}", new DynamicRecordMetadata("${action.name}", Collections.singletonList(new FieldDef<${action.valueClassName}>("value", ${action.valueClassName}.class, DataTemplateUtil.getSchema(${action.valueClassName}.class)))));
+      #else
+        responseMetadataMap.put("${action.name}", new DynamicRecordMetadata("${action.name}", Collections.emptyList()));
+      #end
+    #end
+    _resourceSpec=new ResourceSpecImpl(
+        #if($spec.restMethods.size() > 0)
+          EnumSet.of(
+          #foreach($method in $spec.restMethods)
+              ResourceMethod.${method.method.toUpperCase()}#if($foreach.hasNext),#end
+          #end
+          ),
+        #else
+          EnumSet.noneOf(ResourceMethod.class),
+        #end
+        requestMetadataMap,
+        responseMetadataMap,
+        ${spec.keyClassName}.class,
+        null,
+        null,
+        ${spec.entityClassName}.class,
+        Collections.emptyMap());
+  }
+
+  private final ParSeqRestliClient _client;
+  private final Engine _engine;
+
+  #doc( $spec.resource.doc)
+  public ${spec.className}(ParSeqRestliClient client, Engine engine) {
+    _client = client;
+    _engine = engine;
+  }
+
+  // CRUD methods
```

Comment:
There are excessive blank lines generated here as well

Reply:
These seem to be added by the comments in the templates (copyright). Again, it seems like most template engines have such quirks. Since these are not going to be viewed by users, I think we can ignore those.